yarn/src/test/scala/org/apache/spark/deploy/yarn/BaseYarnClusterSuite.scala

```diff
@@ -116,7 +120,7 @@ abstract class BaseYarnClusterSuite
 
   override def afterAll() {
     yarnCluster.stop()
-    System.clearProperty("SPARK_YARN_MODE")
+    System.setProperties(oldProperties)
     super.afterAll()
```

Comment:
I agree. Since the original code did not use `try finally`, I did not use it, too. Should I use `try finally`?